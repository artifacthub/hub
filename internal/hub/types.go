package hub

type userIDKey struct{}

// UserIDKey represents the key used for the userID value inside a context.
var UserIDKey = userIDKey{}

// ChartRepository represents a Helm chart repository.
type ChartRepository struct {
	ChartRepositoryID string `json:"chart_repository_id"`
	Name              string `json:"name"`
	DisplayName       string `json:"display_name"`
	URL               string `json:"url"`
	UserID            string `json:"user_id"`
}

// Link represents a url associated with a package.
type Link struct {
	Name string `json:"name"`
	URL  string `json:"url"`
}

// Maintainer represents a package's maintainer.
type Maintainer struct {
	MaintainerID string `json:"maintainer_id"`
	Name         string `json:"name"`
	Email        string `json:"email"`
}

// PackageKind represents the kind of a given package.
type PackageKind int64

const (
	// Chart represents a Helm chart.
	Chart PackageKind = 0

	// Falco represents a set of Falco rules.
	Falco PackageKind = 1

	// OPA represents a set of OPA policies.
	OPA PackageKind = 2
)

// Package represents a Kubernetes package.
type Package struct {
	PackageID         string                 `json:"package_id"`
	Kind              PackageKind            `json:"kind"`
	Name              string                 `json:"name"`
	DisplayName       string                 `json:"display_name"`
	Description       string                 `json:"description"`
	HomeURL           string                 `json:"home_url"`
	LogoURL           string                 `json:"logo_url"`
	LogoImageID       string                 `json:"logo_image_id"`
	Keywords          []string               `json:"keywords"`
	Deprecated        bool                   `json:"deprecated"`
	Readme            string                 `json:"readme"`
	Links             []*Link                `json:"links"`
	Version           string                 `json:"version"`
	AvailableVersions []string               `json:"available_versions"`
	AppVersion        string                 `json:"app_version"`
	Digest            string                 `json:"digest"`
	Data              map[string]interface{} `json:"data"`
	Maintainers       []*Maintainer          `json:"maintainers"`
	ChartRepository   *ChartRepository       `json:"chart_repository"`
}

// SearchPackageInput represents the query input when searching for packages.
type SearchPackageInput struct {
	Limit             int           `json:"limit,omitempty"`
	Offset            int           `json:"offset,omitempty"`
	Facets            bool          `json:"facets"`
	Text              string        `json:"text"`
	PackageKinds      []PackageKind `json:"package_kinds,omitempty"`
	ChartRepositories []string      `json:"chart_repositories,omitempty"`
	Deprecated        bool          `json:"deprecated"`
}

// GetPackageInput represents the input used to get a specific package.
type GetPackageInput struct {
	ChartRepositoryName string `json:"chart_repository_name"`
	PackageName         string `json:"package_name"`
	Version             string `json:"version"`
}

// User represents a Hub user.
type User struct {
	UserID        string `json:"user_id"`
	Alias         string `json:"alias"`
	FirstName     string `json:"first_name"`
	LastName      string `json:"last_name"`
	Email         string `json:"email"`
	EmailVerified bool   `json:"email_verified"`
	Password      string `json:"password"`
}

// CheckCredentialsOutput represents the output returned by the
// CheckCredentials method.
type CheckCredentialsOutput struct {
	Valid  bool   `json:"valid"`
	UserID string `json:"user_id"`
}

// Session represents some information about a user session.
type Session struct {
	SessionID string `json:"session_id"`
	UserID    string `json:"user_id"`
	IP        string `json:"ip"`
	UserAgent string `json:"user_agent"`
}

// CheckSessionOutput represents the output returned by the CheckSession method.
type CheckSessionOutput struct {
	Valid  bool   `json:"valid"`
	UserID string `json:"user_id"`
}

// Organization represents an entity with one or more users associated that can
// own packages and other entities like chart repositories.
type Organization struct {
	OrganizationID string `json:"organization_id"`
	Name           string `json:"name"`
	DisplayName    string `json:"display_name"`
	Description    string `json:"description"`
	HomeURL        string `json:"home_url"`
	LogoURL        string `json:"logo_url"`
	LogoImageID    string `json:"logo_image_id"`
}
