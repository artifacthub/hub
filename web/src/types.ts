import { JSONSchema } from '@apidevtools/json-schema-ref-parser';

export enum RepositoryKind {
  Helm = 0,
  Falco,
  OPA,
  OLM,
  TBAction,
  Krew,
  HelmPlugin,
}

export interface Repository {
  repositoryId?: string;
  name: string;
  displayName?: string | null;
  url: string;
  branch?: string | null;
  organizationName?: string | null;
  organizationDisplayName?: string | null;
  userAlias?: string | null;
  kind: RepositoryKind;
  lastTrackingTs?: number | null;
  lastTrackingErrors?: string | null;
  verifiedPublisher?: boolean;
  official?: boolean;
  private?: boolean;
  authUser?: string | null;
  authPass?: string | null;
  disabled?: boolean;
  scannerDisabled?: boolean;
}

export interface Maintainer {
  name?: string;
  email: string;
}

export interface PackageLink {
  name: string;
  url: string;
}

export interface Channel {
  name: string;
  version: string;
}

export interface Package {
  packageId: string;
  name: string;
  displayName: string | null;
  normalizedName: string;
  description: string;
  logoImageId?: string;
  appVersion?: string;
  repository: Repository;
  readme?: string | null;
  install?: string | null;
  data?: PackageData | null;
  availableVersions?: Version[];
  version?: string;
  homeUrl?: string | null;
  keywords?: string[];
  maintainers?: Maintainer[];
  deprecated: boolean | null;
  isOperator?: boolean | null;
  signed: boolean | null;
  links?: PackageLink[];
  stars?: number | null;
  eventKinds?: EventKind[];
  license?: string | null;
  createdAt: number;
  defaultChannel?: string | null;
  channels?: Channel[] | null;
  provider?: string | null;
  containersImages?: ContainerImage[] | null;
  capabilities?: string | null;
  crds?: { [key: string]: any } | null;
  crdsExamples?: CustomResourcesDefinitionExample[] | null;
  securityReportSummary?: SecurityReportSummary | null;
  securityReportCreatedAt?: number;
  hasValuesSchema?: boolean;
  hasChangelog?: boolean;
  contentUrl?: string;
  containsSecurityUpdates?: boolean;
  prerelease?: boolean;
}

export interface ContainerImage {
  image: string;
  name?: string;
}

export interface Version {
  version: string;
  containsSecurityUpdates: boolean;
  prerelease: boolean;
  createdAt: number;
}

export interface OLMExtraData {
  customResourcesExamples?: string;
}

export interface CustomResourcesDefinitionExample {
  kind: string;
  [key: string]: any;
}

export interface CustomResourcesDefinition {
  kind: string;
  name: string;
  version: string;
  description: string;
  displayName?: string;
  example?: CustomResourcesDefinitionExample;
}

export interface Dependency {
  name: string;
  version: string;
  repository?: string;
}

export interface PackageData {
  policies?: OPAPolicies;
  rules?: { Raw: string }[] | FalcoRules;
  dependencies?: Dependency[];
  customResourcesDefinitionsExamples?: string;
  customResourcesDefinitions?: CustomResourcesDefinition[];
  isGlobalOperator?: boolean;
}

export interface OPAPolicies {
  [key: string]: string;
}

export interface FalcoRules {
  [key: string]: string;
}

export interface SearchFiltersURL {
  tsQueryWeb?: string;
  tsQuery?: string[];
  filters: {
    [key: string]: string[];
  };
  pageNumber: number;
  deprecated?: boolean | null;
  operators?: boolean | null;
  verifiedPublisher?: boolean | null;
  official?: boolean | null;
}

export interface SearchQuery {
  tsQueryWeb?: string;
  tsQuery?: string[];
  filters: {
    [key: string]: string[];
  };
  deprecated?: boolean | null;
  operators?: boolean | null;
  verifiedPublisher?: boolean | null;
  official?: boolean | null;
  limit: number;
  offset: number;
  total?: number;
}

export interface SearchResults {
  data: {
    packages: Package[] | null;
    facets: Facets[] | null;
  };
  metadata: {
    limit: number;
    offset: number;
    total: number;
  };
}

export interface Stats {
  packages: number;
  releases: number;
}

export interface Facets {
  filterKey: string;
  title: string;
  options: FacetOption[];
}

export interface FacetOption {
  id: string | number;
  name: string;
  total?: number;
}

export interface Option extends FacetOption {
  filterKey: string;
  icon?: JSX.Element;
}

export interface UserLogin {
  email: string;
  password: string;
}

export interface UserFullName {
  alias: string;
  firstName?: string;
  lastName?: string;
  profileImageId?: null | string;
}

export interface Profile extends UserFullName {
  email: string;
}

export interface User extends UserLogin {
  alias: string;
  firstName?: string;
  lastName?: string;
}

export interface Member {
  alias: string;
  firstName?: string;
  lastName?: string;
  confirmed?: boolean;
}

export interface UserAuth {
  alias?: string;
  status: boolean;
}

export interface CheckAvailabilityProps {
  resourceKind: ResourceKind;
  value: string;
}

export enum ResourceKind {
  userAlias = 'userAlias',
  repositoryName = 'repositoryName',
  repositoryURL = 'repositoryURL',
  organizationName = 'organizationName',
}

export interface Organization {
  name: string;
  displayName?: string;
  homeUrl?: string;
  logoImageId?: string;
  description?: string;
  membersCount?: number | null;
  confirmed?: boolean | null;
}

export interface RefInputField {
  checkIsValid: () => Promise<boolean>;
  reset: () => void;
  getValue: () => string;
  checkValidity: () => boolean;
  updateValue: (value: string) => void;
}

export interface RefInputTypeaheadField {
  reset: () => void;
  getValue: () => string;
  updateValue: (value: string) => void;
}

export interface RefActionBtn {
  reRender: () => void;
}

export interface Alert {
  type: 'success' | 'danger' | 'warning';
  message: string;
  dismissOn?: number;
  autoClose?: boolean;
}

export interface LogoImage {
  imageId: string;
}

export interface AvailabilityInfo {
  isAvailable: boolean;
  resourceKind: ResourceKind;
  excluded: string[];
}

export interface Prefs {
  theme: ThemePrefs;
  controlPanel: {
    selectedOrg?: string;
  };
  search: {
    limit: number;
  };
}

export interface ThemePrefs {
  configured: string;
  efective?: string;
  automatic: boolean;
}

export interface PackageStars {
  starredByUser?: boolean | null;
  stars: number | null;
}

export enum EventKind {
  NewPackageRelease = 0,
  SecurityAlert,
  RepositoryTrackingErrors,
}

export interface Subscription {
  eventKind: EventKind;
}

export interface TestWebhook {
  url: string;
  contentType?: string | null;
  template?: string | null;
  eventKinds: EventKind[];
}

export interface Webhook extends TestWebhook {
  webhookId?: string;
  name: string;
  description?: string;
  secret?: string;
  active: boolean;
  packages: Package[];
  lastNotifications?: null | WebhookNotification[];
}

export interface WebhookNotification {
  notificationId: string;
  createdAt: number;
  processed: boolean;
  processedAt: number;
  error?: null | string;
}

export enum PayloadKind {
  default = 0,
  custom,
}

export interface Section {
  name: string;
  displayName: string;
  shortName?: string;
  disabled: boolean;
  onlyDesktop?: boolean;
  icon?: JSX.Element;
  subsections?: Section[];
}

export interface NavSection {
  user: Section[];
  org: Section[];
}

export interface APIKey {
  apiKeyId?: string;
  name: string;
  createdAt?: number;
}

export interface APIKeyCode {
  key: string;
}

export interface OptionWithIcon {
  value: string;
  label: string;
  icon: JSX.Element;
}

export interface TsQuery {
  name: string;
  label: string;
  value: string;
}
export interface Error {
  kind: ErrorKind;
  message?: string;
}

export enum ErrorKind {
  Other,
  NotFound,
  Unauthorized,
  Forbidden,
}

export interface OptOutItem {
  optOutId: string;
  repository: Repository;
  eventKind: EventKind;
}

export enum AuthorizerAction {
  AddOrganizationMember = 'addOrganizationMember',
  AddOrganizationRepository = 'addOrganizationRepository',
  DeleteOrganization = 'deleteOrganization',
  DeleteOrganizationMember = 'deleteOrganizationMember',
  DeleteOrganizationRepository = 'deleteOrganizationRepository',
  GetAuthorizationPolicy = 'getAuthorizationPolicy',
  TransferOrganizationRepository = 'transferOrganizationRepository',
  UpdateAuthorizationPolicy = 'updateAuthorizationPolicy',
  UpdateOrganization = 'updateOrganization',
  UpdateOrganizationRepository = 'updateOrganizationRepository',
}

export interface AuthorizerInput {
  action: AuthorizerAction;
  organizationName?: string;
  user: string;
  onCompletion?: () => void;
}

export interface RegoPlaygroundPolicy {
  rego_modules: {
    'policy.rego': string;
  };
  input: {
    [key: string]: string | string[];
  };
  data: {
    [key: string]: any;
  };
}

export interface RegoPlaygroundResult {
  result: string;
}

export interface OrganizationPolicy {
  authorizationEnabled: boolean;
  predefinedPolicy?: string | null;
  customPolicy?: string | null;
  policyData?: string | null;
}

export interface AuthorizationPolicy {
  name: string;
  label: string;
  policy: string;
  data: {
    [key: string]: any;
  };
}

export interface SearchTipItem {
  content: JSX.Element | string;
  example: string;
}

export type SecurityReportSummary = {
  [key in VulnerabilitySeverity]?: number;
};

export interface SecurityReport {
  [key: string]: SecurityTargetReport[];
}

export interface SecurityTargetReport {
  Target: string;
  Type: string;
  Vulnerabilities: Vulnerability[] | null;
}

export interface Vulnerability {
  [key: string]: any;
  Severity: VulnerabilitySeverity;
}

export enum VulnerabilitySeverity {
  Critical = 'critical',
  High = 'high',
  Medium = 'medium',
  Low = 'low',
  UnKnown = 'unknown',
}

export interface SeverityRating {
  color: string;
  level: 'A' | 'B' | 'C' | 'D' | 'F' | '-';
}

export type SeverityRatingList = {
  [key in VulnerabilitySeverity | 'default']?: SeverityRating;
};

export interface CVSSVectorMetric {
  value: string;
  label: string;
  options: CVSSVectorOpt[];
}

export interface CVSSVectorOpt {
  value: string;
  label: string;
  level: number;
}

export interface ChangeLog {
  version: string;
  createdAt: number;
  prerelease: boolean;
  containsSecurityUpdates: boolean;
  changes?: string[] | null;
}

export interface ActiveJSONSchemaValue {
  active: number;
  combinationType: string | null;
  options: JSONSchema[];
  error?: boolean;
}
