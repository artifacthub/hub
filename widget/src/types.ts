export interface PackageSummary {
  packageId: string;
  name: string;
  displayName?: string;
  normalizedName: string;
  description: string;
  logoImageId?: string;
  repository: Repository;
  version: string;
  stars?: number | null;
  ts: number;
  official?: boolean;
}

export interface Repository {
  kind: RepositoryKind;
  name: string;
  verifiedPublisher?: boolean;
  official?: boolean;
  organizationName?: string;
  organizationDisplayName?: string;
  userAlias?: string;
}

export enum RepositoryKind {
  Helm = 0,
  Falco,
  OPA,
  OLM,
  TBAction,
  Krew,
  HelmPlugin,
  TektonTask,
  KedaScaler,
  CoreDNS,
  Keptn,
  TektonPipeline,
  Container,
  Kubewarden,
}

export interface SearchResults {
  packages: PackageSummary[] | null;
  facets: null;
}
