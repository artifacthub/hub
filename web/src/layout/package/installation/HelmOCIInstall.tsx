import React from 'react';

import { Repository } from '../../../types';
import { OCI_PREFIX } from '../../../utils/data';
import ExternalLink from '../../common/ExternalLink';
import CommandBlock from './CommandBlock';

interface Props {
  name: string;
  version?: string;
  repository: Repository;
}

const HelmOCIInstall = (props: Props) => {
  const url = props.repository.url.replace(OCI_PREFIX, '');

  return (
    <>
      <CommandBlock command="export HELM_EXPERIMENTAL_OCI=1" title="Enable OCI support" />

      <CommandBlock command={`helm chart pull ${url}:${props.version}`} title="Pull chart from remote" />

      <CommandBlock command={`helm chart export ${url}:${props.version}`} title="Export chart to directory" />

      <CommandBlock command={`helm install my-${props.name} ./${props.name}`} title="Install chart" />

      <div className="mt-2">
        <ExternalLink href="https://helm.sh/docs/intro/quickstart/" className="btn btn-link pl-0">
          Need Helm?
        </ExternalLink>
      </div>
    </>
  );
};

export default HelmOCIInstall;
