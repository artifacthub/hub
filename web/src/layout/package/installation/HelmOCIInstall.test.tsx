import { render } from '@testing-library/react';
import React from 'react';

import { Repository } from '../../../types';
import HelmOCIInstall from './HelmOCIInstall';

const repo: Repository = {
  kind: 0,
  name: 'repo',
  displayName: 'Repo',
  url: 'oci://ghcr.io/artifacthub/artifact-hub',
  userAlias: 'user',
  private: false,
};

const defaultProps = {
  name: 'packageName',
  version: '1.0.0',
  repository: repo,
};

describe('HelmOCIInstall', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const result = render(<HelmOCIInstall {...defaultProps} />);
    expect(result.asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      const { getByText, getAllByText } = render(<HelmOCIInstall {...defaultProps} />);

      expect(getByText('Enable OCI support')).toBeInTheDocument();
      expect(getByText('HELM_EXPERIMENTAL_OCI=1')).toBeInTheDocument();
      expect(getByText('Pull chart from remote')).toBeInTheDocument();
      expect(getAllByText(/ghcr.io\/artifacthub\/artifact-hub:1.0.0/g)).toHaveLength(2);
      expect(getByText('Export chart to directory')).toBeInTheDocument();
      expect(getByText('Install chart')).toBeInTheDocument();
      expect(getByText('helm install my-packageName ./packageName')).toBeInTheDocument();
    });

    it('renders private repo', () => {
      const { getByRole } = render(
        <HelmOCIInstall {...defaultProps} repository={{ ...defaultProps.repository, private: true }} />
      );

      const alert = getByRole('alert');
      expect(alert).toBeInTheDocument();
      expect(alert).toHaveTextContent('Important: This repository is private and requires some credentials.');
    });
  });
});
