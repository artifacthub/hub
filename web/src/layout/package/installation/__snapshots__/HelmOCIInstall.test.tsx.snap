// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelmOCIInstall creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-2"
  >
    <small
      class="text-muted mt-2 mb-1"
    >
      Enable OCI support
    </small>
  </div>
  <div
    class="d-flex align-items-start"
  >
    <div
      class="flex-grow-1 mr-3 blockWrapper"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; color: rgb(0, 0, 0); background: rgb(248, 248, 255);"
      >
        <code
          class="language-bash"
        >
          <span
            style="color: rgb(0, 134, 179);"
          >
            export
          </span>
          <span>
             HELM_EXPERIMENTAL_OCI=1
          </span>
        </code>
      </pre>
    </div>
    <div>
      <div
        class="position-relative undefined"
      >
        <button
          class="btn btn-sm btn-primary rounded-circle copyBtn"
          data-testid="ctcBtn"
          type="button"
        >
          <div
            class="d-flex flex-row align-items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="my-2"
  >
    <small
      class="text-muted mt-2 mb-1"
    >
      Pull chart from remote
    </small>
  </div>
  <div
    class="d-flex align-items-start"
  >
    <div
      class="flex-grow-1 mr-3 blockWrapper"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; color: rgb(0, 0, 0); background: rgb(248, 248, 255);"
      >
        <code
          class="language-bash"
        >
          <span>
            helm chart pull ghcr.io/artifacthub/artifact-hub:1.0.0
          </span>
        </code>
      </pre>
    </div>
    <div>
      <div
        class="position-relative undefined"
      >
        <button
          class="btn btn-sm btn-primary rounded-circle copyBtn"
          data-testid="ctcBtn"
          type="button"
        >
          <div
            class="d-flex flex-row align-items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="my-2"
  >
    <small
      class="text-muted mt-2 mb-1"
    >
      Export chart to directory
    </small>
  </div>
  <div
    class="d-flex align-items-start"
  >
    <div
      class="flex-grow-1 mr-3 blockWrapper"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; color: rgb(0, 0, 0); background: rgb(248, 248, 255);"
      >
        <code
          class="language-bash"
        >
          <span>
            helm chart 
          </span>
          <span
            style="color: rgb(0, 134, 179);"
          >
            export
          </span>
          <span>
             ghcr.io/artifacthub/artifact-hub:1.0.0
          </span>
        </code>
      </pre>
    </div>
    <div>
      <div
        class="position-relative undefined"
      >
        <button
          class="btn btn-sm btn-primary rounded-circle copyBtn"
          data-testid="ctcBtn"
          type="button"
        >
          <div
            class="d-flex flex-row align-items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="my-2"
  >
    <small
      class="text-muted mt-2 mb-1"
    >
      Install chart
    </small>
  </div>
  <div
    class="d-flex align-items-start"
  >
    <div
      class="flex-grow-1 mr-3 blockWrapper"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; color: rgb(0, 0, 0); background: rgb(248, 248, 255);"
      >
        <code
          class="language-bash"
        >
          <span>
            helm install my-packageName ./packageName
          </span>
        </code>
      </pre>
    </div>
    <div>
      <div
        class="position-relative undefined"
      >
        <button
          class="btn btn-sm btn-primary rounded-circle copyBtn"
          data-testid="ctcBtn"
          type="button"
        >
          <div
            class="d-flex flex-row align-items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="mt-2"
  >
    <a
      class="link btn btn-link pl-0"
      href="https://helm.sh/docs/intro/quickstart/"
      rel="noopener noreferrer"
      role="button"
      target="_blank"
    >
      Need Helm?
    </a>
  </div>
</DocumentFragment>
`;
